src/index.ts

```diff
@@ -64,55 +44,71 @@ const getMiniAppButton = (ctx: any) => {
   ENTRYPOINT - START
 */
 bot.start(async (ctx) => {
+  const telegramId = ctx.from?.id;
+
+  // Handle user sign up and get user ID
+  let username: string;
+  const user = await getUserByTelegram(ctx);
+
+  if (!user) {
+    const newUser = await createUserByTelegram(ctx);
+    if (!newUser) {
+      return ctx.reply('Failed to create user.');
+    }
+    username = newUser.username || '';
+  } else {
+    username = user.username || '';
+  }
+
+  // Get all subscribed users
+  const subscribedUsers = await getSubscribedUsers();
+
+  // Auto-subscribe new users
+  if (
+    !subscribedUsers.find((user) => user.telegramUsername === username) &&
+    username
+  ) {
+    await updateUserNotification(ctx, true);
```

Comment:
Why don't we set the notification status in `createUserByTelegram`?

Reply:
Yes! It sets notifications by default for new users. Removed it